### DISCUSSION TITLE
fuzz only left side of an image

### DISCUSSION THREAD

**Discussion by Raffinade** [ImageMagick/ImageMagick#3235] [Help]
Hello everyone, 
i would like to apply the command only on one edge of an image maybe the left side. Is it possible to define an area of 100Pixels from the left edge to remove a special color with white.
Here is my example for the whole image not only for the left side
`for i in *.tif; do convert -fuzz 10% -fill "rgb(255,255,255)" -opaque "rgb(223,199,177)" "$i" "${i%.tif}.TIF" ; done`

Thx
Raffinade

**Comment by fmw42** [comment]
You can simply chop off a certain amount from one side of an image. See https://legacy.imagemagick.org/Usage/crop/#chop or you can trim from one edge using fuzz (only IM 7). See https://imagemagick.org/script/command-line-options.php#trim

**Reply by Raffinade** [reply]
Can you give me please an commandline expamle. This is what i need. 
I think chop is for delete an part of the side of an image, i only want edit one side not to remove it like cropping.

**Reply by fmw42** [reply]
**Just trim the west (left) side of the image**

![rose_b20](https://user-images.githubusercontent.com/13458218/107810707-f4013400-6d21-11eb-93f2-461b923b5af7.png)

`magick rose_b20.png -fuzz 10% -define trim:edges=west -trim +repage rose_b20_trim_west.png`

![rose_b20_trim_west](https://user-images.githubusercontent.com/13458218/107810738-fd8a9c00-6d21-11eb-9cc0-df77e2c79de1.png)

**Comment by GeeMack** [comment]
If you want to modify just a particular area of an image, you should be able to use "-region" to isolate that area using most versions of ImageMagick.

    convert "cardinal.jpg" -fill white -fuzz 20% -region 100x0+0+0 -opaque "srgb(161,187,122)" +region result.png

The selected area is 100 pixels wide and the entire image height, and starts at pixel 0,0. The "-opaque" operation will be applied only to that 100xH+0+0 area. Notice how I added "+region" with the plus "+" after the "-opaque" so any further operations apply to the entire image.

![result](https://user-images.githubusercontent.com/65475279/107812013-b0fc8c00-6d34-11eb-8bd8-978728e0a7e3.png)

**Reply by Raffinade** [reply]
GeeMack this is exactly what i wanted thx soo much ! 
What do i have to do, if i wanted the right side? I have scanned a book all 2nd pages have a shadow on the right side but the pages are not all the same size, so the coordinates WxH+X+Y dont work in the same way for all pages. Is there a way to use things like gravity?
Greetings 
Raffinade

**Reply by GeeMack** [reply]
You should be able to use "-gravity east" to make that same command affect the right side of the image, starting at 100 pixels in from the edge. Be sure to put the "-gravity" setting ahead of the "-region" setting.

**Reply by Raffinade** [reply]
Thx a lot GeeMack! It worked for east:

`for i in *.TIF; do convert -fill white -fuzz 20% -gravity east -region 100x0+0+0 -opaque "srgb(223,199,177)" +region "$i" "${i%.TIF}.tif" ; done`

and it works for west
`for i in *.TIF; do convert -fill white -fuzz 20% -gravity west -region 100x0+0+0 -opaque "srgb(223,199,177)" +region "$i" "${i%.TIF}.tif" ; done`

Greetings and a second thanks
Raffinade :)

**Reply by Raffinade** [reply]
Is it possible to fuzz and to save one color that should not be fuzzed? 
Maybe u want to change a grey against white but at the same time you want to save the color blue (around 10%) in the area where u change grey against white. 

Is this possible?

**Comment by Raffinade** [comment]
Is it possible to change the color in one area for example grey against white and save another color in the same area maybe blue (around 10%). So that blue is not affected by the change to white.

**Reply by GeeMack** [reply]
That sounds like it can be easily done with ImageMagick, but the color modification is somewhat unrelated to this topic. It might be best to put together another question and post it separately. Maybe include an example of what you're trying to achieve and any commands you're working on to get there.

**Reply by Raffinade** [reply]
ok i will do thx
